ts/@live-compositor/core/src/output.ts

```diff
@@ -1,57 +1,91 @@
-import { RegisterOutput, SceneComponent } from 'live-compositor';
+import { ContextStore, LiveCompositorContext, RegisterOutput } from 'live-compositor';
 import { ApiClient, Api } from './api';
 import Renderer from './renderer';
 import { intoAudioInputsConfiguration } from './api/output';
+import React from 'react';
 
 class Output {
   api: ApiClient;
   outputId: string;
   initialized: boolean = false;
 
+  throttledUpdate: () => void;
   videoRenderer?: Renderer;
-  audioOptions?: Api.Audio;
 
-  constructor(outputId: string, output: RegisterOutput, api: ApiClient) {
+  constructor(outputId: string, output: RegisterOutput, api: ApiClient, store: ContextStore) {
     this.api = api;
     this.outputId = outputId;
 
+    let audioOptions: Api.Audio | undefined;
     if (output.video) {
       this.videoRenderer = new Renderer(
-        output.video.root,
-        scene => this.onRendererUpdate(scene),
+        React.createElement(LiveCompositorContext.Provider, { value: store }, output.video.root),
+        () => this.onRendererUpdate(),
         `${outputId}-`
       );
     }
     if (output.audio) {
-      this.audioOptions = intoAudioInputsConfiguration(output.audio.initial);
+      audioOptions = intoAudioInputsConfiguration(output.audio.initial);
     }
-    this.initialized = true;
+    this.throttledUpdate = throttle(async () => {
+      await api.updateScene(this.outputId, {
+        video: this.videoRenderer && { root: this.videoRenderer.scene() },
+        audio: audioOptions,
+      });
+    }, 30);
```

Comment:
Where `30` come from? With 60FPS per frame updates happen every 16ms.

Reply:
> Where 30 come from? 

I think this is optimal value for limiting the updates

> With 60FPS per frame updates happen every 16ms.

I agree with the math, but I don't see what point you are making.

Reply:
With this throttling, it's impossible to do per-frame updates in 60FPS.

Reply:
I don't think it should be possible to do any per frame stuff, there is no way to synchronize it either way